lib/adp/api_connection.rb

```diff
@@ -106,17 +106,21 @@ def get_adp_data(product_url)
                 "redirect_uri" => self.connection_configuration.redirectURL
             };
 
-            data = send_web_request(product_url, data, authorization, 'application/json', 'GET')
+            headers = headers.merge({"Authorization" => authorization})
 
+            response = send_web_request(product_url, data, headers, 'GET')
+
+            # Get parsed body and rescue in case "response.body" is not JSON parsable
+            body = JSON.parse(response.body) rescue {}
             raise ConnectionException, "Connection error: #{data['error']}, #{data['error_description']}" unless data["error"].nil?
```

Comment:
@heliocola Honestly I'm a little bit bummed that I couldn't come up with something smarter here. This is just weird and has "but what if" written all over it. I'll be thinking about it, I'm sure I can come up with a better error handling (well, I just wanted to re-use what ADP gem had already)